ports/lua/CMakeLists.txt

```
@@ -1,117 +1,256 @@
+    "${LUA_SOURCE_DIR}/ldebug.h"
+    "${LUA_SOURCE_DIR}/ldo.h"
+    "${LUA_SOURCE_DIR}/lfunc.h"
+    "${LUA_SOURCE_DIR}/lgc.h"
+    "${LUA_SOURCE_DIR}/llex.h"
+    "${LUA_SOURCE_DIR}/llimits.h"
+    "${LUA_SOURCE_DIR}/lmem.h"
+    "${LUA_SOURCE_DIR}/lobject.h"
+    "${LUA_SOURCE_DIR}/lopcodes.h"
+    "${LUA_SOURCE_DIR}/lparser.h"
+    "${LUA_SOURCE_DIR}/lstate.h"
+    "${LUA_SOURCE_DIR}/lstring.h"
+    "${LUA_SOURCE_DIR}/ltable.h"
+    "${LUA_SOURCE_DIR}/ltm.h"
+    "${LUA_SOURCE_DIR}/lua.h"
+    "${LUA_SOURCE_DIR}/luaconf.h"
+    "${LUA_SOURCE_DIR}/lualib.h"
+    "${LUA_SOURCE_DIR}/lundump.h"
+    "${LUA_SOURCE_DIR}/lvm.h"
+    "${LUA_SOURCE_DIR}/lzio.h"
+)
+
+set(PUBLIC_HDR_LIBLUA
+    "${LUA_SOURCE_DIR}/lualib.h"
+    "${LUA_SOURCE_DIR}/lua.h"
+    "${LUA_SOURCE_DIR}/luaconf.h"
+    "${LUA_SOURCE_DIR}/lauxlib.h"
+    "${LUA_SOURCE_DIR}/lua.hpp"
+)
+
+set(SRC_LIBLUA
+    "${LUA_SOURCE_DIR}/lapi.c"
+    "${LUA_SOURCE_DIR}/lauxlib.c"
+    "${LUA_SOURCE_DIR}/lbaselib.c"
+    "${LUA_SOURCE_DIR}/lbitlib.c"
+    "${LUA_SOURCE_DIR}/lcode.c"
+    "${LUA_SOURCE_DIR}/lcorolib.c"
+    "${LUA_SOURCE_DIR}/lctype.c"
+    "${LUA_SOURCE_DIR}/ldblib.c"
+    "${LUA_SOURCE_DIR}/ldebug.c"
+    "${LUA_SOURCE_DIR}/ldo.c"
+    "${LUA_SOURCE_DIR}/ldump.c"
+    "${LUA_SOURCE_DIR}/lfunc.c"
+    "${LUA_SOURCE_DIR}/lgc.c"
+    "${LUA_SOURCE_DIR}/linit.c"
+    "${LUA_SOURCE_DIR}/liolib.c"
+    "${LUA_SOURCE_DIR}/llex.c"
+    "${LUA_SOURCE_DIR}/lmathlib.c"
+    "${LUA_SOURCE_DIR}/lmem.c"
+    "${LUA_SOURCE_DIR}/loadlib.c"
+    "${LUA_SOURCE_DIR}/lobject.c"
+    "${LUA_SOURCE_DIR}/lopcodes.c"
+    "${LUA_SOURCE_DIR}/loslib.c"
+    "${LUA_SOURCE_DIR}/lparser.c"
+    "${LUA_SOURCE_DIR}/lstate.c"
+    "${LUA_SOURCE_DIR}/lstring.c"
+    "${LUA_SOURCE_DIR}/lstrlib.c"
+    "${LUA_SOURCE_DIR}/ltable.c"
+    "${LUA_SOURCE_DIR}/ltablib.c"
+    "${LUA_SOURCE_DIR}/ltm.c"
+    "${LUA_SOURCE_DIR}/lundump.c"
+    "${LUA_SOURCE_DIR}/lutf8lib.c"
+    "${LUA_SOURCE_DIR}/lvm.c"
+    "${LUA_SOURCE_DIR}/lzio.c"
+)
+
+set_source_files_properties(${SRC_LIBLUA} PROPERTIES LANGUAGE C)
+
+add_library(
+    lua
+    ${SRC_LIBLUA}
+)
+
+if (NOT SKIP_INSTALL_HEADERS)
+    set_target_properties(lua PROPERTIES PUBLIC_HEADER "${PUBLIC_HDR_LIBLUA}")
+endif()
+
+target_include_directories(
+    lua
+    PRIVATE $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/src>
+    PUBLIC $<INSTALL_INTERFACE:${CMAKE_INSTALL_PREFIX}/include>
+)
+
+set_target_properties(
+    lua
+    PROPERTIES
+    POSITION_INDEPENDENT_CODE ON
+)
+
+target_link_libraries(
+    lua
+    ${CMAKE_DL_LIBS}
+)
+
+# Windows specific section
+if (BUILD_SHARED_LIBS AND WIN32)
+    target_compile_definitions(lua PUBLIC LUA_BUILD_AS_DLL)
+endif()
+
+if (WIN32)
+    target_compile_definitions(lua PUBLIC _CRT_SECURE_NO_WARNINGS)
+endif()
+
+# Unix specific section
+if (UNIX)
+    target_compile_definitions(lua PUBLIC LUA_USE_POSIX)
+    # math library
+    find_library(LIB_MATH NAMES m)
+    if (LIB_MATH)
+        target_link_libraries(lua m)
```

To be really "modern", all link statement should use INTERFACE/PUBLIC/PRIVATE.
```suggestion
        target_link_libraries(lua PRIVATE m)
```

---

Yeah fair enough.